pyvisa_py/tcpip.py

```diff
@@ -1273,6 +1273,37 @@ def clear(self) -> StatusCode:
 
         return StatusCode.success
 
+    def flush(self, mask: BufferOperation) -> StatusCode:
+        """Flush the specified buffers.
+        Corresponds to viFlush function of the VISA library.
+        Parameters
+        ----------
+        mask : constants.BufferOperation
+            Specifies the action to be taken with flushing the buffer.
+            The values can be combined using the | operator. However multiple
+            operations on a single buffer cannot be combined.
+        Returns
+        -------
+        constants.StatusCode
+            Return value of the library call.
+        """
+        if (
+            mask & BufferOperation.discard_read_buffer
+            or mask & BufferOperation.discard_read_buffer_no_io
+            or mask & BufferOperation.discard_receive_buffer
+            or mask & BufferOperation.discard_receive_buffer2
```

Comment:
Yes according to the IVI specs this one is should perform IO. I will try to check the serial case.

Reply:
Serial can never receive more than we asked so we not keep an internal buffer on pyvisa side. To me it looks like we may be doing IO when we are not supposed to... Honestly none of this is crystal clear to me. In particular becase we do not implement any of the VISA buffered API.

Reply:
I must omit that my main motivation to get this is to get rid of all pending messages (by doing IO). I thought that that should be `discard_read_buffer` (which for `TCPIP::SOCKET` is equivalent to the `clear` method. And seems to be `reset_input_buffer` for pyserial). 

Maybe it would be better to do nothing in `ASRL` ports on the Buffer operations which are documented to do no IO?
Should this be worked on in this PR?

Reply:
If you want to, feel free to do so. Otherwise it may get buried.

Reply:
I removed the buffer operations which should not perform IO from the flush implementation in serial.